src/redux/saga/product.saga.js

```diff
@@ -0,0 +1,34 @@
+import { put, takeEvery } from "redux-saga/effects";
+
+import {
+  GET_PRODUCT,
+  GET_PRODUCT_FAIL,
+  GET_PRODUCT_SUCCESS,
+} from "../constant";
+import { getProducts } from "../../api/api";
+
+function* getProductSaga(action) {
+  try {
+    const startTime = new Date().getTime();
+
+    const response = yield getProducts(action.payload);
```

Comment:
```suggestion
    const { data } = yield getProducts(action.payload);
```
Hoặc viết rút ngắn gọn bằng destructuring nữa nhé
```
const { data: payload } = yield getProducts(action.payload);
yield put({
    type: GET_PRODUCT_SUCCESS,
    payload,
});
```